view/adminhtml/web/js/validation.js

```diff
@@ -0,0 +1,22 @@
+require([
+        'jquery',
+        'mage/translate',
+        'jquery/validate'],
+    function($){
+        $.validator.addMethod(
+            "reach-default-hs-code-validator", function (hscode) {
+                //Got help/hints from
+                //https://magento.stackexchange.com/questions/95171/magento-2-form-validation
+                //https://magento.stackexchange.com/questions/139023/custom-field-validation-in-system-xml
+                //https://regex101.com/
+                //This routine is enforcing good enough check on length so we possibly can remove the built in
+                // validation of the form `validate-length minimum-length-2 maximum-length-20` from system.xml.
+                //Does accept two consecutive special characters. Also allows mix and match of `.` and `-` in the
+                //hscode string - made it that way after talking (on Slack) to the team.
+                //Check is done only at frontend level
+                //Additionally the check could be done/enforced when getting saved in db (in a path way bypassing UI)
+                hscode = hscode.trimRight();
+                return (/^[\d]{2}[A-Za-z0-9\.-]{0,18}$/.test(hscode));
+            }, $.mage.__('Please enter a string that starts with two digits and have at least 2 characters and at most 20 characters (letter and numbers)'));
```

Comment:
if this is the error message if this validation doesn't work, could explicitly say "Please enter an HS Code that starts..." because that's what the field is? Or would you prefer this error message be more vague ("a string")

Reply:
I would love to put a better error message that you suggested. The message is something I came up with but it sounds more developer oriented than end user oriented (I agree)   😄 Thanks Lizzy!

Reply:
haha for sure! Thank you :)

Reply:
Commit 4d5773a1339a054f289c08c348e8d9d7b9e4aacf (in addition to addressing your concerns used something similar to the second approach described below)

Thanks Lizzy!  @lizzynibali